### ISSUE TITLE
Enhance CRD guideline with owner referencing

### ISSUE THREAD

**Issue by a-thaler** [kyma-project/community#43]
Confirm these statements before you submit the issue:

- [x] I have searched open and closed issues for duplicates.
- [x] I have read the contributing guidelines.
---

**Description**
Currently it is hard to figure out what resources were created by controllers for Kyma CRDs. For example the API controller creates an ingress resource but it is hard to identify that it belongs to an API resource (you need to know the naming convention). 

Without an easy way to recognize system resources that are managed by controllers (creation and deletion) we have the following issues
* Cannot provide filters on the UI to hide that "system" resources in an easy way 
* Provide transparency to the end users (e.g. developers) about which resources are generated/managed by Kyma

**Suggestion**
whenever a kyma controller creates a resource in consequence of a kyma CRD resource, that resource will be enriched with all needed ownerReference attributes, see [kubernetes guide](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection). That needs to be documented in the guide for CRD development.

If done, please think about creating follow up tickets for adjusting existing controllers.

**Comment by Abd4llA** [comment]
@brendandobbs I think this can be looked as a "Kyma developer experience" issue, no? if so we can add it to the quality roadmap and priortize it accordingly.

**Comment by abbi-gaurav** [comment]
Post upgrading Kyma to use K8S version 1.11, it should be possible to use the k8s status sub-resources feature for custom resources.

**Comment by pbochynski** [comment]
I think that we should use categories instead of labels. See my ticket: https://github.com/kyma-project/kyma/issues/2780

**Comment by a-thaler** [comment]
Good idea to introduce categories. However, an additional labelling can bring benefits for example to indicate also by which resource it is managed in detail. Also other tooling might be mainly based on labels like prometheus.
But maybe it is not so much about labelling sub-resource but introducing a common and agreed set of kyma labels helping to identify components/modules and types of them as well as helping in finding things. Think of uninstall where you want to delete all resources belonging to the installation after installer was uninstalling everything aleady)

**Comment by pbochynski** [comment]
I think for removing stuff we should use owner references: https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#owners-and-dependents
I am not against labels, just wanted to ensure that we use first concepts that are used in the kubernetes ecosystem before we introduce own solution based on labels.

**Comment by brendandobbs** [comment]
As there are potentially different requirements @Disper will setup a call to brainstorm before we can come up with a proposal.

**Comment by mszostok** [comment]
I cannot join the discussion, but I'm also with PB, we should first use the common concepts and if they are not sufficient then come up with sth more.

- categories on CRDs 
- proper usage of OwnerReference
- and about the labels see this doc: https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels
  > You can visualize and manage Kubernetes objects with more tools than kubectl and the dashboard. A common set of labels allows tools to work interoperably, describing objects in a common manner that all tools can understand.

**Comment by a-thaler** [comment]
I agree, the ownerReferences concept will solve my initial thought, I updated the description accordingly.

**Comment by derberg** [comment]
Please help in prioritizing the issue and decide if it is important to focus on it now

**Comment by lilitgh** [comment]
The issue is still valid.
Consulted with @a-thaler . It's a good practice and something nice to have but it's not critical.

**Comment by pbochynski** [comment]
We should follow kubernetes guidelines and we don't need a special guide for Kyma.
